src/lib/transition_handler/processor.ml

```diff
@@ -84,7 +84,7 @@ let add_and_finalize ~logger ~frontier ~catchup_scheduler
 let process_transition ~logger ~trust_system ~verifier ~frontier
     ~catchup_scheduler ~processed_transition_writer ~time_controller
     ~transition:cached_initially_validated_transition ~precomputed_values =
-  let transition_receipt_time = Time.now () in
+  let transition_receipt_time = Some (Time.now ()) in
```

Comment:
Missed this last review, but this is *not* the receipt time. This can actually be a long while after the receipt time, as the transitions sit in a queue to be handled synchronously before here. The correct way to capture the receipt time would be to keep the transition in the envelope up until this point (or, if that's too hard, pass it as a tuple of the transition and the time it was received or something).

Reply:
I think the transition here is a cached enveloped. Envelopes don't have a timestamp, as far as I see. Maybe that's worth adding, which would make extracting the timestamp simple here.

Reply:
You're right, the Envelope doesn't have a timestamp atm. My recommendation would be to add one, and set that when we build the Envelope.Incoming from the networking layer.